### PR TITLE
Add example tests.

### PR DESCRIPTION
**Pull request by avietrov**
This PR adds a few tests to the reference implementation.

Should be merged after #16

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,21 +14,21 @@
     <dependency>
       <groupId>com.netflix.hollow</groupId>
       <artifactId>hollow</artifactId>
-      <version>2.6.8</version>
+      <version>4.7.1</version>
```
Comment (rydenius) [inline pom.xml:17]: Jump all the way to 4.7.12 and use a variable so the hollow version does not need to be repeated?

File: src/test/java/how/hollow/ExampleTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    MoviePrimaryKeyIndex index = new MoviePrimaryKeyIndex(consumer);
+    assertNull(index.findMatch(newMovie.id));
+
+    announceNewVersion(2L, new HollowWriteStateEngineBuilder()
+        .add(movie)
+        .add(newMovie)
+        .build());
+
+    assertNotNull(index.findMatch(newMovie.id));
+    assertEquals(newMovie.title, index.findMatch(newMovie.id).getTitle().getValue());
```
Comment (rydenius) [inline src/test/java/how/hollow/ExampleTest.java:142]: Add asserts for `index.findMatch(movie.id)` not being null both before and after the second version?

File: src/test/java/how/hollow/ExampleTest.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  public void movieIndexUpdate() throws Exception {
+
+    Movie movie = new Movie(1, "title", Set.of());
+    Movie newMovie = new Movie(2, "coming soon", Set.of());
+
+    announceNewVersion(latestVersion, new HollowWriteStateEngineBuilder()
+        .add(movie)
+        .build());
+
+    MoviePrimaryKeyIndex index = new MoviePrimaryKeyIndex(consumer);
```
Comment (liufuyang) [inline src/test/java/how/hollow/ExampleTest.java:131]: Isn't this deprecated now?

In my example I can see that I could basically use something like :
```
// Setting up an id index
val uniqueIndex = Movie.uniqueIndex(consumer)
consumer.addRefreshListener(uniqueIndex)
```
The `Movie` type is the generated class.

File: pom.xml (hunk 1 above)
Comment (avietrov) [reply pom.xml:17]: I've forked from @liufuyang 's forked where the version bump was introduced, the PR for that #16 is still open

Comment (avietrov) [reply pom.xml:17]: Done in the other RP :)

File: src/test/java/how/hollow/ExampleTest.java (hunk 2 above)
Comment (avietrov) [reply src/test/java/how/hollow/ExampleTest.java:142]: Done too

File: src/test/java/how/hollow/ExampleTest.java (hunk 3 above)
Comment (avietrov) [reply src/test/java/how/hollow/ExampleTest.java:131]: Yes, I'll fix that as soon as I have your PR merged and then I can use the new API.

**Pull request closed without merging**